### PR TITLE
feature/8-dataCreationUI

### PR DESCRIPTION
**Pull request by Jaakik**

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/DataCreationActivity.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    public fun removeView(view:View){
+        binding.parentLinearLayout.removeView(view.parent as View)
+
+    }
+
+    // this method will be the building block for linking this activtiy to the camera activity
+
+    /*private fun saveData() {
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/DataCreationActivity.kt:53]: commented-out code is a bad practice, either uncomment it or remove it

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/DataCreationActivity.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        super.onCreate(savedInstanceState)
+        _binding = ActivityDataCreationBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.buttonAdd.setOnClickListener {
+            addNewView()
+        }
+
+
+        /*binding.buttonSubmitList.setOnClickListener {
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/DataCreationActivity.kt:31]: commented-out code is a bad practice, either uncomment it or remove it

**Review by bbjubjub2494 — APPROVED** [review]

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/DataCreationActivity.kt (hunk 1 above)
Comment (Jaakik) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/DataCreationActivity.kt:53]: Yes you are right, I  just wanted to leave it out as is later for Jonas when trying to link both activities.

File: app/src/main/res/layout/activity_data_creation.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_margin="15dp"
+        android:text="@string/DataCreationTitle"
```
Comment (jonaskon) [inline app/src/main/res/layout/activity_data_creation.xml:15]: change to `android:text="@string/DataCreationActivity_DataCreationTitle"`

File: app/src/main/res/layout/activity_data_creation.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        android:textColor="@color/black"
+        android:textSize="15sp" />
+
+
+    <LinearLayout
+        android:id="@+id/parent_linear_layout"
```
Comment (jonaskon) [inline app/src/main/res/layout/activity_data_creation.xml:21]: change to `android:id="@+id/DataCreationActivity_parent_linear_layout"

File: app/src/main/res/layout/activity_data_creation.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical" />
+
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/button_add"
```
Comment (jonaskon) [inline app/src/main/res/layout/activity_data_creation.xml:27]: Change to `android:id="@+id/DataCreationActivity_button_add"`

File: app/src/main/res/layout/activity_data_creation.xml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        android:layout_marginRight="15dp"
+        android:paddingLeft="10dp"
+        android:paddingRight="10dp"
+        android:text="@string/DataCrationSubmitButton"
+        android:textAllCaps="false"
+        android:textColor="@color/white" />
+
+
+</LinearLayout>
+
```
Comment (jonaskon) [inline app/src/main/res/layout/activity_data_creation.xml:55]: Prefix all ids and strings with "DataCreationActivity" to make it easier to manage the resources.
e.g `android:text="@string/DataCrationSubmitButton"` to `android:text="@string/DataCreationActivity_DataCrationSubmitButton"`

File: app/src/main/res/layout/row_add_category.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        android:background="#BF1D1D"
+        android:onClick="removeView"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/et_name"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.538"
+        app:srcCompat="@android:drawable/ic_menu_delete" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+
```
Comment (jonaskon) [inline app/src/main/res/layout/row_add_category.xml:34]: Same as https://github.com/HumanLearning2021/HumanLearningApp/pull/26/files#r597118546

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -7,4 +7,8 @@
     <string name="AddPicture_categorySelectionDialogTitle">Pick a category</string>
     <string name="AddPicture_permissionNeededDialogTitle">Camera required</string>
     <string name="AddPicture_permissionNeededDialogMessage">To use this functionality, access to the camera is required. Either you have denied the app the permission to use it or your device has no camera.</string>
+    <string name="DataCreationTitle">Categories List</string>
+    <string name="DataCrationSubmitButton">Submit</string>
+    <string name="DataCreationAddButton">Add</string>
+    <string name="DataCreationCat">Enter Category</string>
 </resources>
```
Comment (wattsecs) [inline app/src/main/res/values/strings.xml]: Prefix all strings with "DataCreationActivity" to make it easier to manage the resources.
e.g `android:text="@string/DataCrationSubmitButton"` to `android:text="@string/DataCreationActivity_DataCrationSubmitButton"`

**Review by wattsecs — CHANGES_REQUESTED** [review]

**Review by wattsecs — APPROVED** [review]

**Pull request merged**